skyportal/tests/api/test_annotations.py

```diff
@@ -0,0 +1,296 @@
+import numpy.testing as npt
+from skyportal.tests import api
+
+
+def test_post_without_origin_fails(annotation_token, public_source, public_group):
+
+    # this should not work, since no "origin" is given
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'data': {'redshift': 0.15},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status == 400
+    assert data["message"] == "Missing required field `origin`"
+
+    # this should not work, since "origin" is empty
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'origin': '',
```

Comment:
The empty string should be allowed as a valid origin. It should only barf when origin is null

Reply:
I'm not sure about this. If people are allowed to enter empty origin it could cause a lot of trouble... Part of the whole annotations API is that you MUST give an origin when posting these. If '' is legal how is that enforced?

Reply:
`''` is not an empty origin. It is the empty string, which is a valid string. If you want to require that an origin string consist of at least 1 alphanumeric character then that is not what is being validated in the current handler. For example, I could post `' '` and the handler as written would accept that as valid user input. That is fine. As long as it is a unique string, that is OK. We should, to the extent possible, let data types specify valid contracts with the users.

Reply:
I see your point. But if we don't limit anything (maybe 1+ alphanumeric is good) then it leaves the option for abusing the system. If I'm writing a bot and I don't have patience to read the definition of "origin" I could just leave it empty. Or put a " " or a "-" as input until it works and I wouldn't care if that is actually not informative. 

I think we should enforce some reasonable constraint on this. It is easier than "educating" the users to be good and properly fill out their origin...